client/my-sites/plans/main.jsx

```diff
@@ -37,12 +38,30 @@ const Plans = React.createClass( {
 	},
 
 	componentDidMount() {
+		this.redirectIfNonJetpackMonthly();
+
 		// Scroll to the top
 		if ( typeof window !== 'undefined' ) {
 			window.scrollTo( 0, 0 );
 		}
 	},
 
+	componentDidUpdate() {
```

Comment:
Wondering if using `componentWillReceiveProps()` would be slightly better here so we can redirect before rendering?

Reply:
I don't think it would make a big difference, and I usually try to avoid having side effects in `componentWillReceiveProps`.

Reply:
I'm fine to change it if you have a hard opinion on it though.

Reply:
> I don't think it would make a big difference

Yes, I agree there, I don't think it matters much, so I'm fine the way it is.

> I usually try to avoid having side effects in componentWillReceiveProps

Curious about this, though. The [docs](https://facebook.github.io/react/docs/react-component.html#componentwillreceiveprops) don't mention avoiding side effects. `componentWillReceiveProps` is the earliest opportunity to take action on changed props, so why wait till later?

Reply:
Oh yeah, I totally agree it's not included in the docs, and it's probably something that should be mentioned there. Sadly I can only provide rough non-docs links for consideration, like this one for example: https://twitter.com/acdlite/status/818931319703044096?lang=en